sub/lavc_conv.c

```diff
@@ -82,14 +81,30 @@ struct lavc_conv *lavc_conv_create(struct mp_log *log,
     avctx = avcodec_alloc_context3(codec);
     if (!avctx)
         goto error;
-    if (mp_set_avctx_codec_headers(avctx, mp_codec) < 0)
+    if (mp_set_avctx_codec_headers(avctx, sd->codec) < 0)
         goto error;
 
     priv->avpkt = av_packet_alloc();
     priv->avpkt_vtt = av_packet_alloc();
     if (!priv->avpkt || !priv->avpkt_vtt)
         goto error;
 
+    switch (codec->id) {
+    case AV_CODEC_ID_DVB_TELETEXT:;
+        if (!sd->opts->teletext_page) {
```

Comment:
I changed my mind, will move this to decode(). Just so we can update page number during runtime. I think there is value in that.

Reply:
done